x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -0,0 +1,368 @@
+            .build();
+    }
+
+    private static XContentBuilder getConnectorsIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                builder.field("version", Version.CURRENT.toString());
+                builder.endObject();
+
+                builder.field("dynamic", "false");
+                builder.startObject("properties");
+                {
+                    builder.startObject(Connector.ERROR_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.INDEX_NAME_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.IS_NATIVE_FIELD.getPreferredName());
+                    builder.field("type", "boolean");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LANGUAGE_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SEEN_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SYNC_ERROR_FIELD.getPreferredName());
+                    builder.field("type", "text");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SYNC_STATUS_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SYNC_SCHEDULED_AT_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SYNCED_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(Connector.NAME_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.SERVICE_TYPE_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.STATUS_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.SYNC_NOW_FIELD.getPreferredName());
+                    builder.field("type", "boolean");
+                    builder.endObject();
+                }
+                builder.endObject();
```

Comment:
This object looks to be missing:
- `api_key_id`,
- `configuration`
- `custom_scheduling`
- `description`
- `features`
- `filtering`
- `last_deleted_document_count`
- `last_indexed_document_count`
- `pipeline`
- `scheduling`

Reply:
That's on purpose (also for the other missing mappings). I'll send out an email about this, but there's no real point to adding these as mappings--they bloat the index, but they're not searchable.

Reply:
We still need to declare them with:
```
"configuration": {
        "type": "object",
        "enabled": false
      }
```
That way we can set the mapping to be strict and not accept any other field in updates.

Reply:
> they bloat the index, but they're not searchable

We also want to make sure they aren't indexed as `text`, which I think is the default for any non-declared field.